website/frontend/src/components/blocks/Wallet/Wallet.tsx

```diff
@@ -1,28 +1,25 @@
-import React, { useEffect, useRef, useState } from 'react';
+import { useEffect, useState } from 'react';
 import clsx from 'clsx';
 import Identicon from '@polkadot/react-identicon';
 import type { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
-import { GearKeyring } from '@gear-js/api';
-import { useAlert } from 'react-alert';
-import { LogoutIcon } from 'assets/Icons';
-import { useApi } from 'hooks';
-import { Modal } from '../Modal';
-import { AccountList } from './AccountList';
-import { nodeApi } from '../../../api/initApi';
-import { LOCAL_STORAGE } from 'consts';
-import { useAccount } from 'hooks';
-import { useAccounts } from './hooks';
+import { UnsubscribePromise } from '@polkadot/api/types';
+import { Button, buttonStyles } from '@gear-js/ui';
+
 import styles from './Wallet.module.scss';
+import { useAccounts } from './hooks';
+import { SelectAccountModal } from './SelectAccountModal';
 
-export const Wallet = () => {
-  const injectedAccounts = useAccounts();
-  const [accountBalance, setAccountBalance] = useState<string | null>(null);
-  const [isOpen, setIsOpen] = useState(false);
+import { LOCAL_STORAGE } from 'consts';
+import { useAccount, useApi } from 'hooks';
 
-  const alert = useAlert();
+const Wallet = () => {
   const { api } = useApi();
+  const injectedAccounts = useAccounts();
```

Comment:
```suggestion
  const accounts = useAccounts();
```
`useAccounts` executes `web3enable` polkadot api, therefore I would suggest to use it only once there, and then pass accounts as a prop to modal